server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -229,6 +230,9 @@ public DateTime(DateFormatter formatter, ZoneId timeZone, DateFieldMapper.Resolu
         }
 
         private DateTime(DateFormatter formatter, ZoneId timeZone, DateFieldMapper.Resolution resolution, boolean formatSortValues) {
+            if (formatter.pattern().equals("epoch_second") || formatter.pattern().equals("epoch_millis")) {
```

Comment:
@pgomulka is there a better way to detect if we have an epoch-based format?  Some kind of `isEpochBased()` method or similar?